### PR TITLE
Fix arch is undefined, update tests, and allow manual invocation of action

### PR DESCRIPTION
**Pull request by r-LaForge**
Fix arch is undefined, update tests, and allow manual invocation of action.

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -51,7 +51,9 @@ test-run:
     RUN ! grep 'Found tool in cache' output
     RUN cat output | grep '^::add-path::' | sed 's/::add-path:://g' > earthly-path
     RUN test "$(cat earthly-path)" = "/root/.earthly/bin"
-    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | grep '^earthly version v.*linux/amd64; Linux'
+    # [a-zA-Z0-9]* attempt to match a commit hash
+    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version
+    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | grep '^earthly version v.* [a-zA-Z0-9]* linux/amd64; Alpine Linux'
```
Comment (r-LaForge) [inline Earthfile:56]: Running as a separate command simply makes it easier to see the version to make the necessary changes. This test relies on the specific format of our version command.

File: dist/setup/index.js [hunk 2]
```diff
@@ -68324,7 +68324,8 @@ function run() {
                 throw new Error(`Unsupported operating system - ${pkgName} is only released for ${Object.keys(nodePlatformToReleasePlatform).join(", ")}`);
             }
             const releasePlatform = nodePlatformToReleasePlatform[runnerPlatform];
-            const releaseArch = nodeArchToReleaseArch[os.arch()];
+            const osArch = os.arch();
+            const releaseArch = nodeArchToReleaseArch[os.arch()] || osArch;
```
Comment (r-LaForge) [inline dist/setup/index.js:68328]: When the osArch wasn't in the map we got `undefined` and cached `linux-undefined`.

File: src/setup-earthly.ts [hunk 3]
```diff
@@ -37,7 +37,8 @@ async function run() {
     }
 
     const releasePlatform = nodePlatformToReleasePlatform[runnerPlatform];
-    const releaseArch = nodeArchToReleaseArch[os.arch()];
+    const osArch = os.arch();
+    const releaseArch = nodeArchToReleaseArch[os.arch()] || osArch;
```
Comment (r-LaForge) [inline src/setup-earthly.ts:41]: When the osArch wasn't in the map we got undefined and cached linux-undefined.

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 node_modules
 coverage
+.idea
```
Comment (alexcb) [inline .gitignore:3]: nit: https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/

File: Earthfile (hunk 1 above)
Comment (alexcb) [inline Earthfile:56]: it looks like this regex now requires two spaces for cases like `v0.1.2<space><space>linux/amd64`

should it be something closer to:
```suggestion
    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | grep '^earthly version v.*( [a-zA-Z0-9]+)? linux/amd64; Alpine Linux'
```

However note that there's already a `.*` should should match everything between `v` and `linux/amd64`, so I'm not sure about this regex change.

What about:

```
RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | tee version.output
RUN grep '^earthly version v.*linux/amd64; Linux' version.output
```

instead?

File: src/lib/__tests__/__snapshots__/get-version.test.ts.snap [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
+exports[`get-version range versions should match 0.*.* versions 1`] = `"v0.6.23"`;
 
 exports[`get-version range versions should match 0.4.* versions 1`] = `"v0.4.6"`;
 
 exports[`get-version range versions should match 0.6.1 versions 1`] = `"v0.6.1"`;
 
 exports[`get-version range versions should match 0.6.1 versions 2`] = `"v0.6.1"`;
 
-exports[`get-version range versions should match latest versions 1`] = `"v0.6.14"`;
+exports[`get-version range versions should match latest versions 1`] = `"v0.6.23"`;
```
[the remark this answers is not in the document]
Comment (alexcb) [reply src/lib/__tests__/__snapshots__/get-version.test.ts.snap:13]: ouch!

Maybe we should parse the string into `[0, 6, 23]` then we can just programatically check that it's greater than `[0,6,1]`.

**Review by alexcb — COMMENTED** [review]
looking good!

I left a couple of small suggestions to make this more manageable in the future.

File: Earthfile (hunk 1 above)
Comment (r-LaForge) [reply Earthfile]: This is very close. 
I've adjusted the regex you've provided just a bit to allow for the specific flavour of Linux as well (Alpine in this example).

```
RUN grep '^earthly version v.*linux/amd64; .* Linux' version.output

```

File: src/lib/__tests__/__snapshots__/get-version.test.ts.snap (hunk 5 above)
[the remark this answers is not in the document]
Comment (r-LaForge) [reply src/lib/__tests__/__snapshots__/get-version.test.ts.snap]: I decided to use `semver` for this. Made the tests very straightforward.

File: Earthfile [hunk 6]
```diff
@@ -51,7 +51,9 @@ test-run:
     RUN ! grep 'Found tool in cache' output
     RUN cat output | grep '^::add-path::' | sed 's/::add-path:://g' > earthly-path
     RUN test "$(cat earthly-path)" = "/root/.earthly/bin"
-    RUN export PATH="$(cat earthly-path):$PATH" && earthly --version | grep '^earthly version v.*linux/amd64; Linux'
+    # [a-zA-Z0-9]* attempt to match a commit hash
```
Comment (alexcb) [inline Earthfile:54]: is this comment still relevant?

**Review by alexcb — APPROVED** [review]

**Pull request merged**